src/main/java/org/opensearch/flowframework/model/Template.java

```diff
@@ -53,6 +54,14 @@ public class Template implements ToXContentObject {
     public static final String TEMPLATE_FIELD = "template";
     /** The template field name for template use case */
     public static final String USE_CASE_FIELD = "use_case";
+    /** Fields which may be updated in the template even if provisioned */
+    private static final Set<String> UPDATE_FIELD_ALLOWLIST = Set.of(
+        NAME_FIELD,
+        DESCRIPTION_FIELD,
+        USE_CASE_FIELD,
+        VERSION_FIELD,
+        UI_METADATA_FIELD
+    );
```

Comment:
Can we copy/paste this allowlist on the issue to keep the track on?

Reply:
[Done](https://github.com/opensearch-project/flow-framework/issues/749#issuecomment-2214960669)